sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -470,6 +470,15 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
     case _ => false
   }
 
+  private def isAlwaysFalse(exps: Seq[Expression], equalTo: Literal): Boolean = {
```

Comment:
`isAlwaysFalse` seems to be mismatched a little with the behavior of this function. Technically, this function resembles `contains` function where `expr` is a list and `equalTo` literal is the item to be check the containment.
```scala
scala> Seq(1, 2).contains(3)
res0: Boolean = false

scala> Seq(1, 2).contains(2)
res1: Boolean = true
```

Reply:
Yes, do you have a recommended function name?

Reply:
Maybe, something like `notEqualForAll` instead of `isAlwaysFalse`?